### PR TITLE
[P1-L15] Prevent Governor proposals with data from being sent to EOA accounts

### PR DESCRIPTION
**Pull request by chrismaree**
The current implementation enables a transaction proposal intended for a smart contract (has data) to be sent to an EOA. This is not desired as this transaction will fail silently within the internal `_executeCall` function. 

This PR ensures that if there is data within a proposed transaction, it is intended to be sent to a contract and not an EOA.

### CODE REVIEW

File: core/contracts/oracle/implementation/Governor.sol [hunk 1]
```diff
@@ -94,6 +96,10 @@ contract Governor is MultiRole, Testable {
         // Initialize the transaction array.
         for (uint256 i = 0; i < transactions.length; i++) {
             require(transactions[i].to != address(0), "The to address cannot be 0x0");
+            // If the transaction has any data with it the recipient must be a contract, not an EOA.
+            if (transactions[i].data.length > 1) {
```
Comment (mrice32) [inline core/contracts/oracle/implementation/Governor.sol:100]: Shouldn't this be `>0`?

Comment (chrismaree) [reply core/contracts/oracle/implementation/Governor.sol:100]: We represent no data like this: `web3.utils.hexToBytes("0x0")`. This has a length of 1, not zero. It's a bit nasty. I can add a comment

Comment (mrice32) [reply core/contracts/oracle/implementation/Governor.sol:100]: Is there a way to pass in a bytes array of length 0?

Comment (chrismaree) [reply core/contracts/oracle/implementation/Governor.sol]: Ye, that's a good point. I changed this to use `web3.utils.hexToBytes("0x")` which is a bytes array of length 0. I have updated the contracts accordingly to check for any length `>0` for a tx with data. is cleaner now, thanks for the suggestion 👍.

File: core/test/oracle/Governor.js [hunk 2]
```diff
@@ -238,7 +254,7 @@ contract("Governor", function(accounts) {
       {
         to: account2,
         value: amountToDeposit,
-        data: web3.utils.hexToBytes("0x0")
+        data: web3.utils.hexToBytes("0x")
```
Comment (chrismaree) [inline core/test/oracle/Governor.js:257]: We now use this to represent zero byte data.

**Pull request merged**